virttest/data_dir.py

```diff
@@ -214,8 +216,16 @@ def get_tmp_dir(public=True):
 
 
 def get_download_dir():
-    if not os.path.isdir(DOWNLOAD_DIR):
-        shutil.copytree(BASE_DOWNLOAD_DIR, DOWNLOAD_DIR)
+    """
+    Returns the dir that holds download definition files, AKA asset files
+
+    Before retuning the download directory, it makes sure the user, writable
+    copy of the download dir is created, and that all asset definition
+    file are in sync.
+
+    :returns: a writable directory populated with asset definition files
+    """
+    dir_util.copy_tree(BASE_DOWNLOAD_DIR, DOWNLOAD_DIR)
```

Comment:
I agree that the whole concept is awkward, but I'm reluctant to redesign this at this time.

What do you think if we only get newer files from the systemwide download dir?  This is slightly different from what you propose because, if users modify their local files, there will be no overwrite (theirs newer than ours).  But, if we ship a newer version of the same file, or a brand new file, than the user's copy would be overwritten (ours newer than theirs).

If users creates a new file, than it would never be overwritten. IMHO, it makes sense that users should create *new* resource files, and not really change the ones we ship.

The proposal is based on the fact that this would be a very easy fix.  Again, I'm reluctant to change too much here at this time.

What do you think?